File: Copilot/2.Estructura de Datos Pila (Stack)/golang/pila.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strings"
)

type Stack struct {
    items []string
}

func (s *Stack) Push(item string) {
    s.items = append(s.items, item)
}

func (s *Stack) Pop() string {
    if len(s.items) == 0 {
        return "Pila vacía"
    }
    item := s.items[len(s.items)-1]
    s.items = s.items[:len(s.items)-1]
    return item
}

func (s *Stack) IsEmpty() bool {
    return len(s.items) == 0
}

func main() {
    stack := Stack{}
    reader := bufio.NewReader(os.Stdin)

    for {
        fmt.Print("Elige una opción (push, pop, salir): ")
        op, _ := reader.ReadString('\n')
        op = strings.TrimSpace(op)

        if op == "push" {
            fmt.Print("Valor a agregar: ")
            val, _ := reader.ReadString('\n')
            stack.Push(strings.TrimSpace(val))
        } else if op == "pop" {
            fmt.Println("Elemento eliminado:", stack.Pop())
        } else if op == "salir" {
            break
        }
    }
}

File: Copilot/2.Estructura de Datos Pila (Stack)/Typescript/pila.ts
class Stack<T> {
    private items: T[] = [];

    push(element: T): void {
        this.items.push(element);
        console.log(`Elemento ${element} agregado.`);
    }

    pop(): T | undefined {
        const removed = this.items.pop();
        console.log(`Elemento eliminado: ${removed}`);
        return removed;
    }

    isEmpty(): boolean {
        return this.items.length === 0;
    }

    print(): void {
        console.log("Contenido de la pila:", this.items.join(", "));
    }
}

// Simulación de interacción
const stack = new Stack<string>();
let option: string | null;

do {
    const input = prompt("Opciones: push, pop, print, salir");
    option = input ? input.toLowerCase() : null;
    switch (option) {
        case "push":
            const value = prompt("Ingresa un valor:");
            if (value) stack.push(value);
            break;
        case "pop":
            stack.pop();
            break;
        case "print":
            stack.print();
            break;
    }
} while (option !== "salir");
